crypto/x509/v3_asid.c

```diff
@@ -299,8 +299,7 @@ static int ASIdentifierChoice_is_canonical(ASIdentifierChoice *choice)
         if ((bn == NULL && (bn = BN_new()) == NULL) ||
             ASN1_INTEGER_to_BN(a_max, bn) == NULL ||
             !BN_add_word(bn, 1) ||
-            (a_max_plus_one =
-             BN_to_ASN1_INTEGER(bn, a_max_plus_one)) == NULL) {
+            BN_to_ASN1_INTEGER(bn, a_max_plus_one) == NULL) {
```

Comment:
no this can't work. if a_max_plus_one is NULL this is an allocation.

Reply:
Ah, I didn't notice that possibility.  I'll fix.

Reply:
see #9833

Reply:
Dropping this part of this PR in favour of that.

Reply:
(which I just approved).